apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx

```diff
@@ -18,7 +16,16 @@ export default class DetailViewContents extends React.Component {
       email: PropTypes.string,
       form_data: PropTypes.object
     }),
-    updateProps: PropTypes.func.isRequired
+    updateProps: PropTypes.func.isRequired,
+    path: PropTypes.string.isRequired
+  };
+
+  statuses = () => {
+    let statuses = ['Unreviewed', 'Pending', 'Waitlisted', 'Accepted', 'Declined', 'Withdrawn'];
+    if (this.props.path.includes('facilitators')) {
+      statuses.splice(2, 0, 'Interview');
+    }
```

Comment:
Is there a reason this `DetailViewContents` needs to know its path? It seems it just needs to know whether it's a teacher or facilitator view, which would be better accomplished with an enum.

I'd prefer this prop to be something like `view_type: PropTypes.oneOf('teacher', 'facilitator').isRequired`, and move the path parsing into the caller.

Zooming out, I also don't like taking this dependency on the path anyway, even if it's moved out to the caller. I'd rather this be more declarative. Since we already have a [hash](https://github.com/code-dot-org/code-dot-org/blob/4c1364745b902a6e3d731d0364eef07cf056b360/apps/src/code-studio/pd/application_dashboard/application_dashboard.jsx#L30-L36) of all the various paths that we're using to [construct the routes](https://github.com/code-dot-org/code-dot-org/blob/4c1364745b902a6e3d731d0364eef07cf056b360/apps/src/code-studio/pd/application_dashboard/application_dashboard.jsx#L56). Why not add more data here, instead of mapping directly from path to name, do something like 
```jsx
csf_facilitators: {type: 'facilitators', course: 'csf', name: 'CS Fundamentals Facilitator Applications'}
```

And then pass those props directly into the `DetailView` routes.

Reply:
That is a great idea, and will also allow me to clean up some things in quick view. thanks!